src/lib/swiss.js

```diff
@@ -100,148 +106,57 @@ function getStandings(options, round, participants, matches) {
-    return standings.findIndex(el => el.id === a.home) -
-      standings.findIndex(el => el.id === b.home)
-  })
-  return matchups
-}
-
-function generatePenalties(options, orderedParticipants, matches) {
-  var penalties = orderedParticipants.map((participant, index) => {
-    var opponents = orderedParticipants.slice()
-    opponents.splice(index, 1)
-    var penalties = opponents.map((opponent, idx) => {
-      var penalty = matches.filter(match => {
-        return (
-          match.home.id === participant.id &&
-          match.away.id === opponent.id
-        ) || (
-          match.home.id === opponent.id &&
-          match.away.id === participant.id
+  var arr = []
+  mappings.map(team => {
+    mappings.map(opp => {
+      if(team.id !== opp.id) {
+        arr.push([
+          team.id,
+          opp.id,
+          -1 * (Math.pow(team.points - opp.points, options.standingPower) +
+          options.rematchWeight*team.opponents.reduce((n, o) => {
+            return n + (o === opp.id)
+          }, 0))]
         )
-      }).length * options.rematchWeight
-      penalty += ((index - idx) > 0 ? index - idx - 1 : idx - index) *
-        options.standingWeight
-      penalty += Math.abs(participant.seed - opponent.seed)
-      return {
-        id: opponent.id,
-        penalty: penalty
+      } else {
+        arr.push([team.id, opp.id, -10000])
       }
     })
-    return {
-      id: participant.id,
-      penalties: penalties
-    }
   })
-  return penalties.reduce((acc, penaltyList) => {
-    acc[penaltyList.id] = penaltyList.penalties.reduce((acc, penalty) => {
-      acc[penalty.id] = penalty.penalty
-      return acc
-    }, {})
-    return acc
-  }, {})
-}
 
-function* getAllPairs(participants) {
-  if (participants.length < 2) {
-    yield [participants]
-    return
-  }
-
-  var a = participants[0]
-  participants = participants.slice(1)
-  for (var [i, val] of participants.entries()) {
-    var pair = [a, val]
-    var others = participants.slice(0, i).concat(participants.slice(i+1))
-    if (others.length === 0) {
-      yield [pair]
-    } else {
-      for (var rest of getAllPairs(others)) {
-        yield [pair].concat(rest)
-      }
+  var results = blossom(arr, true)
+  var matchups = []
+  for(var i = 0; i < results.length; ++i) {
+    if(results[i] !== -1 && !results.reduce((n, r) => { return n + (+r.away === i)}, 0)) {
+      matchups.push({'penalty': NaN, 'home': i, 'away': results[i]})
     }
   }
+  return matchups
 }
 
 module.exports = (options) => {
-  if (!options) {
-    options = {}
-  }
-  if (!options.maxPerRound) {
-    options.maxPerRound = 1
-  }
-  if (!options.rematchWeight) {
-    options.rematchWeight = 1000000
-  }
-  if (!options.standingWeight) {
-    options.standingWeight = 10000
-  }
+  options = options || {}
+  options.maxPerRound = options.maxPerRound || 1
+  options.rematchWeight = options.rematchWeight || 10
+  options.standingPower = options.standingPower || 2
```

Comment:
All we really need is absolute value which powers provide for free.  We can adjust the power based on how impactful you think 1 or more points are.